### PR TITLE
[FEATURE] Alignment Configuration: put align_cfg::debug to align_cfg::detail::debug

### PR DESCRIPTION
**Pull request by Irallia**
Resolves https://github.com/seqan/product_backlog/issues/181

Blocked by #2033

### CODE REVIEW

**Comment by Irallia** [comment]
I get some documentation errors:
```bash
/Users/lydia/Repos/seqan3/include/seqan3/alignment/configuration/detail.hpp:29: warning: unable to resolve reference to 'seqan3::align_cfg::detail::debug' for \ref command
/Users/lydia/Repos/seqan3/include/seqan3/alignment/all.hpp:92: warning: unable to resolve reference to 'seqan3::align_cfg::detail::debug' for \ref command
/Users/lydia/Repos/seqan3/include/seqan3/alignment/configuration/detail.hpp:29: warning: unable to resolve reference to 'seqan3::align_cfg::detail::debug' for \ref command
```
Does someone know how to solve them? Do I need to add a config detail struct?

File: include/seqan3/alignment/configuration/align_config_aligned_ends.hpp [hunk 1]
```diff
@@ -245,10 +245,10 @@ back_end_second(value_t) -> back_end_second<value_t>;
...
 //!\cond
     requires (sizeof...(ends_t) <= 4) &&
-             ((detail::is_type_specialisation_of_v<ends_t, front_end_first> ||
-               detail::is_type_specialisation_of_v<ends_t, back_end_first> ||
-               detail::is_type_specialisation_of_v<ends_t, front_end_second>  ||
-               detail::is_type_specialisation_of_v<ends_t, back_end_second>) && ...)
+             ((seqan3::detail::is_type_specialisation_of_v<ends_t, front_end_first> ||
+               seqan3::detail::is_type_specialisation_of_v<ends_t, back_end_first> ||
+               seqan3::detail::is_type_specialisation_of_v<ends_t, front_end_second>  ||
+               seqan3::detail::is_type_specialisation_of_v<ends_t, back_end_second>) && ...)
```
Comment (marehr) [inline include/seqan3/alignment/configuration/align_config_aligned_ends.hpp:251]: This change seems unrelated

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,8 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
 * We lowered the requirements of the `seqan3::aligned_sequence` concept by removing everything that needs write
   access to the object. We then added a new `seqan3::writable_aligned_sequence` concept which extends
   `seqan3::aligned_sequence` by the requirements that need write access (e.g. `insert_gap`).
+* We put some `align_cfg` elements into a subnamespace `align_cfg::detail`:
+  `align_cfg::detail::debug` ([\#2020](https://github.com/seqan/seqan3/pull/2020)).
```
Comment (marehr) [inline CHANGELOG.md:46]: @seqan/core Do we want to add a CHANGELOG entry for moved entities from `seqan3::align_cfg` namespace to `seqan3::align_cfg::detail`?

Comment (smehringer) [reply CHANGELOG.md:46]: hm should the user know about the `debug` functionality at all? If not, I wouldn't put it into the changelog

Comment (Irallia) [reply CHANGELOG.md:46]: Yesterday I discussed this with @joergi-w because he uses many of our detailed functions for his app. The debug_stream, for example, is very practical.

Comment (marehr) [reply CHANGELOG.md:46]: The debug_stream is not detail, it's public API.

https://github.com/seqan/seqan3/blob/c24c911880d350e3388616563ad73b9ca9ac0ba7/include/seqan3/alignment/configuration/align_config_debug.hpp#L19-L38

`seqan3::align_cfg::debug` is semi-public and only used within tests. If it is the only change, I wouldn't mention it. (I only cursorily looked over the PR)

**Comment by marehr** [comment]
@Irallia I pushed directly i your branch to fix the last things. Can you rebase and re-order the history?

The name-clash commits changes should be before adding the namespace.

File: CHANGELOG.md (hunk 2 above)
Comment (rrahn) [reply CHANGELOG.md:46]: No the debug thing is developer API only. And as far as I know it was also only documented like so. Which means the regular user does not know about it. That's why we decided to put it into a specific detail sub namespace.

> Yesterday I discussed this with @joergi-w because he uses many of our detailed functions for his app. The debug_stream, for example, is very practical.

That is the responsibility of the app developer. We don't give any guarantees about the detail namespace. That can change at any time without notice.

**Comment by marehr** [comment]
@Irallia can you rebase?
